### PR TITLE
Feature/health check

### PR DESCRIPTION
**Pull request by loneil**
# Description
Health check, refactoring, api check tweaks.

## Types of changes

New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/ConfigGeneratedJson.vue [hunk 1]
```diff
@@ -1,12 +1,14 @@
 <template>
   <v-container>
-    <v-textarea
-      auto-grow
-      readonly
-      label="Application Configuration"
-      v-model="appConfigAsString"
-      class="jsonText"
-    ></v-textarea>
+    <div class="">
```
Comment (jujaga) [inline frontend/src/components/ConfigGeneratedJson.vue]: Does class need to be specified when it's empty?

File: frontend/src/components/HealthCheck.vue [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                </v-flex>
+              </v-layout>
+              <v-layout row>
+                <v-flex xs6>Top level endpoint return</v-flex>
+                <v-flex xs6>
+                  <span v-html="this.getStatus(healthCheck.endpoints[1].healthCheck)"></span>
+                </v-flex>
+              </v-layout>
+            </div>
+          </v-flex>
```
Comment (jujaga) [inline frontend/src/components/HealthCheck.vue]: Lines 12-60 could be rewritten as a v-for to handle more cases - should we consider changing that API endpoint to also contain the name of the API endpoint as well?

File: frontend/src/components/Home.vue [hunk 3]
```diff
@@ -21,6 +21,50 @@
...
+                  <v-icon>healing</v-icon>
+                </v-btn>
+              </template>
+              <span>API Health Check</span>
+            </v-tooltip>
+
+            <v-dialog v-model="dialog" width="600">
+              <v-card>
+                <v-card-title class="headline primary white--text" primary-title>
+                  <v-icon dark right>healing</v-icon>&nbsp; API Health Check
```
Comment (jujaga) [inline frontend/src/components/Home.vue]: I'm presuming there's no alternative to using the `&nbsp;` for this?

File: frontend/src/components/HealthCheck.vue (hunk 2 above)
Comment (loneil) [reply frontend/src/components/HealthCheck.vue]: Lets do that. I will adjust the api endpoint and yaml docs

File: frontend/src/components/ConfigGeneratedJson.vue (hunk 1 above)
Comment (loneil) [reply frontend/src/components/ConfigGeneratedJson.vue]: No

File: frontend/src/components/Home.vue (hunk 3 above)
Comment (loneil) [reply frontend/src/components/Home.vue]: Adjusting to use a spacer instead

**Pull request merged**